exercises/list-ops/canonical-data.json

```diff
@@ -186,16 +314,80 @@
           "description": "direction dependent function applied to non-empty list",
           "property": "foldl",
           "input": {
-            "list": [2, 5],
+            "list": [
+              2,
+              5
+            ],
             "initial": 5,
             "function": "(x, y) -> x / y"
           },
           "expected": 0
+        },
+        {
+          "uuid": "36549237-f765-4a4c-bfd9-5d3a8f7b07d2",
+          "reimplements": "613b20b7-1873-4070-a3a6-70ae5f50d7cc",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input."
+          ],
+          "description": "empty list",
+          "property": "foldl",
+          "input": {
+            "list": [],
+            "initial": 2,
+            "function": "(acc, el) -> el * acc"
+          },
+          "expected": 2
+        },
+        {
+          "uuid": "7a626a3c-03ec-42bc-9840-53f280e13067",
+          "reimplements": "e56df3eb-9405-416a-b13a-aabb4c3b5194",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input."
+          ],
+          "description": "direction independent function applied to non-empty list",
+          "property": "foldl",
+          "input": {
+            "list": [
+              1,
+              2,
+              3,
+              4
+            ],
+            "initial": 5,
+            "function": "(acc, el) -> el + acc"
+          },
+          "expected": 15
+        },
+        {
+          "uuid": "d7fcad99-e88e-40e1-a539-4c519681f390",
+          "reimplements": "d2cf5644-aee1-4dfc-9b88-06896676fe27",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input."
+          ],
+          "description": "direction dependent function applied to non-empty list",
+          "property": "foldl",
+          "input": {
+            "list": [
+              1,
+              2,
+              3,
+              4
+            ],
+            "initial": 24,
+            "function": "(acc, el) -> el / acc"
+          },
+          "expected": 64
```

Comment:
I see that this is correct, though I'd like to note for my records that it needs the `/` to preserve fractions. Integer division will not work here, since it would compute `1 / 24 = 0`. I will take it into account when using this case.

Reply:
Yes! I should add that as a comment.

I think it was the same with the previous case?

Reply:
I think the previous case asked for integer division, since its expected value is zero and the only way to get a zero out of division is if you started with a zero. Since there is no zero present in the list nor in the initial value, I reason that the zero must have been introduced via integer division.

Reply:
Right. I couldn't get the test case to work in some languages I work in _because_ they do fractional division. I think commenting here is the right approach, and something we should have had on the previous case. I _will_ add it there too!